sky/clouds/cloud.py

```diff
@@ -485,5 +486,25 @@ def query_status(cls, name: str, tag_filters: Dict[str, str],
         """
         raise NotImplementedError
 
+    @classmethod
+    def create_image_from_cluster(cls, name: str, tag_filters: Dict[str, str],
+                                  region: Optional[str],
+                                  zone: Optional[str]) -> str:
+        raise NotImplementedError
+
+    @classmethod
+    def maybe_move_image(cls, image_name: str, source_region: str,
+                         target_region: str, source_zone: Optional[str],
+                         target_zone: Optional[str]) -> str:
+        """Move an image if the target region cannot find it.
+
+        Returns: the image ID in the target region.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    def delete_image(cls, image_id: str, region: Optional[str]) -> None:
+        raise NotImplementedError
```

Comment:
We should add docstrs to all 3 methods, and:

- Say it's used by `sky launch --clone-disk-from`
- Say if it's ok for a cloud impl to leave these out. If so, they should add CloudImplementationFeatures.xx to the non supported list.